---
title: C6312
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6312
helpviewer_keywords:
- C6312
ms.assetid: 1fc8b9a1-e6ba-4799-84c3-31f289576cca
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 48f18934d1bf78a1764bd3ac7a7c8b7b64508ece
ms.sourcegitcommit: 535ef05b1e553f0fc66082cd2e0998817eb2a56a
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 10/07/2019
ms.locfileid: "72017762"
---
# <a name="c6312"></a>C6312
Warnung C6312: Mögliche unendliche Schleife: Verwendung der Konstanten EXCEPTION_CONTINUE_EXECUTION im Ausnahme Filter Ausdruck eines Try-außer

 Diese Warnung gibt die Verwendung der Konstanten `EXCEPTION_CONTINUE_EXECUTION` (oder eine andere Konstante, die zu-1 ausgewertet wird) im Filter Ausdruck eines strukturierten Ausnahme Handlers an. Die Verwendung des konstanten Werts `EXCEPTION_CONTINUE_EXECUTION` kann zu einer Endlosschleife führen. Wenn z. b. eine Ausnahme von der Hardware ausgelöst wurde, wird die Anweisung, die die Ausnahme verursacht hat, neu gestartet. Wenn die Adresse, die die Ausnahme verursacht hat, immer noch schlecht ist, wird eine andere Ausnahme auftreten und auf die gleiche Weise behandelt. Dies verursacht eine Endlosschleife.

 Ein expliziter Rückruf von `RaiseException` führt nicht direkt zu einer Endlosschleife, sondern wird die Ausführung des Codes im geschützten Block fortsetzen. Dies kann unerwartet sein und zu einer Endlosschleife führen, wenn `RaiseException` verwendet wurde, um das Dereferenzieren eines ungültigen Zeigers zu vermeiden.

 In der Regel sollte `EXCEPTION_CONTINUE_EXECUTION` nur von einer im Filter Ausdruck aufgerufenen Funktion zurückgegeben werden, die die Möglichkeit hat, entweder den Zeiger zu korrigieren, der die Ausnahme verursacht hat, oder den zugrunde liegenden Arbeitsspeicher.

## <a name="example"></a>Beispiel
 Der folgende Code generiert diese Warnung:

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
   *ptr = '\0';
  }
  __except (EXCEPTION_CONTINUE_EXECUTION)
    // When EXCEPTION_CONTINUE_EXECUTION is used, the handler
    //  block of the structured exception handler is not executed.
  {
    puts("This block is never executed");
  }
}
```

 So korrigieren Sie die Warnung unter Verwendung des folgenden Codes

```
#include <excpt.h>
#include <stdio.h>
#include <windows.h>

void f (char *ptr)
{
  __try
  {
    // exception occurs if the caller passes null ptr
    // code...
    *ptr = '\0';
  }
  __except (GetExceptionCode()== EXCEPTION_ACCESS_VIOLATION ?
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)
  {
    puts("Error Occurred");
  }
}
```